.github/workflows/build-and-deploy.yml

```
@@ -2,16 +2,79 @@ name: Build & Deploy
+          
+          # Check if any commit message doesn't follow Conventional Commits
+          if git log --oneline -n 1 | grep -qvE '^[a-f0-9]{7} (feat|fix|docs|style|refactor|test|chore)(\(.+\))?:'; then
+            echo "❌ Commit message does not follow Conventional Commits format"
+            # exit 1 # Keep it for warning during initial phase
+          fi
+          echo "✅ Commit message format validated"
+
 permissions:
   contents: read
 
 jobs:
```

actionlint (1.7.10)

[error] key "jobs" is duplicated in "workflow" section. previously defined at line:23,col:1

(syntax-check)